addons/controllers/calico/calicoconfig_controller.go

```diff
@@ -47,47 +48,45 @@ type CalicoConfigReconciler struct {
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *CalicoConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.Log = r.Log.WithValues("CalicoConfig", req.NamespacedName)
+	log := r.Log.WithValues("CalicoConfig", req.NamespacedName)
```

Comment:
In other places r.Log is used. Since this is changed to log the other places where r.Log is used it is going to not log calicoconfig req.NamespacedName. It is important to know the name and namespace of calicoconfig object in the logs.

If this is unnecessary please keep it as r.Log

Reply:
this causes a race condition Test failure. (reasigning r.log =r.log) if you note other  providers use  log = r.log (use log). will try again, but the build fails with race

Reply:
currently testing to see if we can get away with using r.log=r.log

Reply:
yes this fails if we use r.Log = r.Log. 
see https://github.com/vmware-tanzu/tanzu-framework/actions/runs/3388165981/jobs/5629815922#step:15:1305
WARNING: DATA RACE
Write at 0x00c0012c8290 by goroutine 158:
  github.com/vmware-tanzu/tanzu-framework/addons/controllers/calico.(*CalicoConfigReconciler).Reconcile()
      /home/runner/work/tanzu-framework/tanzu-framework/addons/controllers/calico/calicoconfig_controller.go:51 +0x1cd
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:121 +0x16a
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:320 +0x488
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273 +0x324
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234 +0xc8

Previous read at 0x00c0012c8290 by goroutine 359:
  runtime.racereadrange()
      <autogenerated>:1 +0x1b
  github.com/vmware-tanzu/tanzu-framework/addons/controllers/calico.(*CalicoConfigReconciler).ClusterToCalicoConfig-fm()
      <autogenerated>:1 +0x4d
  sigs.k8s.io/controller-runtime/pkg/handler.(*enqueueRequestsFromMapFunc).mapAndEnqueue()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/handler/enqueue_mapped.go:80 +0x84
  sigs.k8s.io/controller-runtime/pkg/handler.(*enqueueRequestsFromMapFunc).Create()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/handler/enqueue_mapped.go:57 +0xd1
  sigs.k8s.io/controller-runtime/pkg/source/internal.EventHandler.OnAdd()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/internal/eventsource.go:63 +0x311
  sigs.k8s.io/controller-runtime/pkg/source/internal.(*EventHandler).OnAdd()
      <autogenerated>:1 +0xd5
  k8s.io/client-go/tools/cache.(*processorListener).run.func1()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.24.6/tools/cache/shared_informer.go:818 +0x151
  k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:155 +0x48
  k8s.io/apimachinery/pkg/util/wait.BackoffUntil()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:156 +0xce
  k8s.io/apimachinery/pkg/util/wait.JitterUntil()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:133 +0x104
  k8s.io/apimachinery/pkg/util/wait.Until()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:90 +0x78
  k8s.io/client-go/tools/cache.(*processorListener).run()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.24.6/tools/cache/shared_informer.go:812 +0x18
  k8s.io/client-go/tools/cache.(*processorListener).run-fm()
      <autogenerated>:1 +0x39
  k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:73 +0x73

Goroutine 158 (running) created at:
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:230 +0x4a4
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:241 +0x2a8
  sigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/manager/runnable_group.go:219 +0x1b8
  sigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func2()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/manager/runnable_group.go:222 +0x47

Goroutine 359 (running) created at:
  k8s.io/apimachinery/pkg/util/wait.(*Group).Start()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.24.6/pkg/util/wait/wait.go:71 +0xdc
  k8s.io/client-go/tools/cache.(*sharedProcessor).addListener()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.24.6/tools/cache/shared_informer.go:634 +0x379
  k8s.io/client-go/tools/cache.(*sharedIndexInformer).AddEventHandlerWithResyncPeriod()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.24.6/tools/cache/shared_informer.go:560 +0x63d
  k8s.io/client-go/tools/cache.(*sharedIndexInformer).AddEventHandler()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.24.6/tools/cache/shared_informer.go:496 +0x54
  sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1()
      /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/source.go:158 +0x310
==================

Reply:
Got it. Thanks for checking out

Reply:
Also run into this issue in other util func